manifests/init.pp

```diff
@@ -71,111 +71,115 @@
+  $tlscafile                                           = $zabbix::params::server_tlscafile,
+  $tlscertfile                                         = $zabbix::params::server_tlscertfile,
+  $tlscrlfile                                          = $zabbix::params::server_tlscrlfile,
+  $tlskeyfile                                          = $zabbix::params::server_tlskeyfile,
+  $trappertimeout                                      = $zabbix::params::server_trappertimeout,
+  $unreachableperiod                                   = $zabbix::params::server_unreachableperiod,
+  $unavailabledelay                                    = $zabbix::params::server_unavailabledelay,
+  $unreachabledelay                                    = $zabbix::params::server_unreachabledelay,
+  $alertscriptspath                                    = $zabbix::params::server_alertscriptspath,
+  $externalscripts                                     = $zabbix::params::server_externalscripts,
+  $fpinglocation                                       = $zabbix::params::server_fpinglocation,
+  $fping6location                                      = $zabbix::params::server_fping6location,
+  $sshkeylocation                                      = $zabbix::params::server_sshkeylocation,
+  $logslowqueries                                      = $zabbix::params::server_logslowqueries,
+  $tmpdir                                              = $zabbix::params::server_tmpdir,
+  $startproxypollers                                   = $zabbix::params::server_startproxypollers,
+  $proxyconfigfrequency                                = $zabbix::params::server_proxyconfigfrequency,
+  $proxydatafrequency                                  = $zabbix::params::server_proxydatafrequency,
+  $allowroot                                           = $zabbix::params::server_allowroot,
+  $include_dir                                         = $zabbix::params::server_include,
+  $loadmodulepath                                      = $zabbix::params::server_loadmodulepath,
+  $loadmodule                                          = $zabbix::params::server_loadmodule,
+  Optional[Stdlib::Absolutepath] $socketdir            = $zabbix::params::server_socketdir,
+  Boolean $manage_selinux                              = $zabbix::params::manage_selinux,
+  String $additional_service_params                    = $zabbix::params::additional_service_params,
+  Optional[String[1]] $zabbix_user                     = $zabbix::params::server_zabbix_user,
+  Optional[String] $zabbix_server_name                 = $zabbix::params::zabbix_server,
+  Optional[Stdlib::Absolutepath] $saml_sp_key          = $zabbix::params::saml_sp_key,
+  Optional[Stdlib::Absolutepath] $saml_sp_cert         = $zabbix::params::saml_sp_cert,
+  Optional[Stdlib::Absolutepath] $saml_idp_cert        = $zabbix::params::saml_idp_cert,
+  Optional[Hash[String[1], ScalarData]] $saml_settings = $zabbix::params::saml_settings,
```

Comment:
can you instead default to an empty hash? we try to use that as default instead of undef. we've our code guidelines at https://voxpupuli.org/docs/reviewing_pr/

Reply:
Thanks, I'll go through the guidelines!
Is `Optional[Hash]` type okay for this purpose?

Reply:
no, it should be `Hash[String[1], ScalarData]` and the value in params.pp should be `{}` and not undef

Reply:
Mhmm that doesn't work, because I can have a hash as a value. 

https://github.com/voxpupuli/puppet-zabbix/pull/763/files#diff-82ccb785b8e760dc28ab7067a48ebd387c93d71c401fbebb9256bfa38117014aR221-R229

Reply:
why doesn't that work? It's a common pattern to default to an empty hash. The datatypes with in the `Hash[]` type only apply if the hash is not empty, so that's fine in this case. Do the tests throw any error?

Reply:
Defaulting to empty hash is okay.

This is the error: 

```
Class[Zabbix::Web]: parameter 'saml_settings' entry 'security' expects a ScalarData value, got Struct
```

Reply:
`Hash[String[1], Variant[ScalarData, Hash]]` seems to be working